### PR TITLE
Add option to export patterns

### PR DESCRIPTION
**Pull request by philip-cline**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR facilitates the export of the proprietary `datatools_patterns.txt` file, introduced in https://github.com/conveyal/gtfs-lib/pulls, which allows the preservation of custom pattern names in the editor. 

Front end PR: https://github.com/ibi-group/datatools-ui/compare/add-option-for-publishing-patterns?expand=1
GTFS-lib PR: https://github.com/conveyal/gtfs-lib/pull/394

### CODE REVIEW

**Review by br648 — APPROVED** [review]
LGTM. Will be good to get this merged so dev is in-sync with GTFS-lib dev which already contains this work.

File: src/main/java/com/conveyal/datatools/editor/controllers/api/SnapshotController.java [hunk 1]
```diff
@@ -63,6 +63,12 @@ private static Snapshot getSnapshotFromRequest(Request req) {
         return Persistence.snapshots.getById(id);
     }
 
+    private static boolean getPublishProprietaryFiles(Request req) {
+        return Boolean.parseBoolean(
+            req.queryParamOrDefault("publishProprietaryFiles",Boolean.FALSE.toString())
```
Comment (miles-grant-ibigroup) [inline src/main/java/com/conveyal/datatools/editor/controllers/api/SnapshotController.java]:
```suggestion
            req.queryParamOrDefault("publishProprietaryFiles", Boolean.FALSE.toString())
```
Missing space

File: src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java [hunk 2]
```diff
@@ -130,7 +130,8 @@ public void jobLogic() {
             snapshot.feedTransformResult = dbTarget.feedTransformResult;
             // If the user has selected to create a new version from the resulting snapshot, do so here.
             if (rules.createNewVersion) {
-                addNextJob(new CreateFeedVersionFromSnapshotJob(feedSource, snapshot, owner));
+                // Publishing the proprietary files will preserve the pattern names in the newly created feed version.
+                addNextJob(new CreateFeedVersionFromSnapshotJob(feedSource, snapshot, owner, true));
```
Comment (miles-grant-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java:134]: This feels like a scary new default. How are we sure this won't have weird side effects?

**Review by miles-grant-ibigroup — APPROVED** [review]
Very clean, just one major q

File: src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java (hunk 2 above)
Comment (philip-cline) [reply src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java:134]: This is a safe default bc the method is only used to process incoming GTFS files. All the export methods allow for user selection if they want to export the pattern names and this just allows re-import

**Pull request merged**